lib/cf_deployer/stack.rb

```diff
@@ -127,8 +127,8 @@ def update_stack(template, params, capabilities, tags, override_policy_json)
       unless override_policy_json.nil?
         args[:stack_policy_during_update_body] = override_policy_json
       end
-      @cf_driver.update_stack(template, args)
-      wait_for_stack_op_terminate
+      stack_updated = @cf_driver.update_stack(template, args)
+      wait_for_stack_op_terminate if stack_updated
```

Comment:
just to be clear, why is this really needed?  "stack_updated" here means "stack update routine ran"
supposing it didn't run this time?, why does `wait_for stack_op_terminate` have to be guarded from this?
I'm guessing because if it failed *last* time, `wait_for stack_op_terminate` return a failure here, and that's wrong.